lib/src/piece/case.dart

```diff
@@ -89,22 +81,18 @@ class CaseExpressionPiece extends Piece {
     // If there is a split guard, then indent the pattern past it.
     var indentPatternForGuard = !_canBlockSplitPattern &&
         !_patternIsLogicalOr &&
-        (state == _beforeWhen || state == _beforeWhenAndBody);
+        state == _beforeWhenAndBody;
 
     if (indentPatternForGuard) writer.pushIndent(Indent.expression);
 
-    writer.pushAllowNewlines(allowNewlineInPattern);
-    writer.format(_pattern);
-    writer.popAllowNewlines();
+    writer.format(_pattern, allowNewlines: allowNewlineInPattern);
 
     if (indentPatternForGuard) writer.popIndent();
 
     if (_guard case var guard?) {
       writer.pushIndent(Indent.expression);
-      writer.pushAllowNewlines(allowNewlineInGuard);
```

Comment:
...on this line `allowNewlineInGuard` will be `false` when `state` is `_beforeWhen`. So this solution will be invalidated if a newline is written...